File: src/Host/OperationsSystem.Blazor/OperationsSystem.Blazor.Client/Auth/AuthModels.cs
namespace OperationsSystem.Blazor.Client.Auth;

public sealed record AccessTokenResponse(string AccessToken, DateTimeOffset ExpiresAtUtc);

public sealed record LoginResponse(bool MfaRequired, string? MfaToken, string? AccessToken, DateTimeOffset? ExpiresAtUtc);

public sealed record LoginOutcome(bool MfaRequired, string? MfaToken)
{
    public static LoginOutcome SignedIn { get; } = new(false, null);
    public static LoginOutcome RequiresMfa(string token) => new(true, token);
}

public sealed record AuthenticatedUser(
    Guid Id,
    string Email,
    string DisplayName,
    Guid RoleId,
    string RoleName,
    string UserType,
    Guid? ExternalReferenceId,
    string PortalSource,
    bool MfaEnabled,
    bool MfaEnrollmentRequired,
    IReadOnlyList<string> Permissions);

public sealed record LoginRequest(string Email, string Password);
public sealed record LoginMfaRequest(string MfaToken, string Code);

File: src/Host/OperationsSystem.Mobile/app/src/main/java/com/nags/operations/data/AuthModels.kt
package com.nags.operations.data

import kotlinx.serialization.Serializable

/** Mirrors `LoginRequest` on the backend (`Identity.Api.Endpoints.IdentityRequests`). */
@Serializable
data class LoginRequest(
    val email: String,
    val password: String,
)

/**
 * Response of `POST /api/v1/identity/auth/mobile/login`. The server answers with either the
 * token set (no MFA) or an MFA challenge (`mfaRequired` + `mfaToken`); the nullable fields let
 * one DTO cover both shapes.
 */
@Serializable
data class MobileLoginResponse(
    val accessToken: String? = null,
    val accessTokenExpiresAtUtc: String? = null,
    val refreshToken: String? = null,
    val refreshTokenExpiresAtUtc: String? = null,
    val mfaRequired: Boolean = false,
    val mfaToken: String? = null,
)

/** Mirrors `MobileTokensResponse`: the refresh token travels in the JSON body for mobile. */
@Serializable
data class MobileTokensResponse(
    val accessToken: String,
    val accessTokenExpiresAtUtc: String,
    val refreshToken: String,
    val refreshTokenExpiresAtUtc: String,
)

@Serializable
data class LoginMfaRequest(val mfaToken: String, val code: String)

@Serializable
data class RefreshRequest(val refreshToken: String)

@Serializable
data class MobileLogoutRequest(val refreshToken: String?)

/** Mirrors `AuthenticatedUserDto` from `GET /api/v1/identity/me`. */
@Serializable
data class AuthenticatedUser(
    val id: String,
    val email: String,
    val displayName: String,
    val userType: String,
    val externalReferenceId: String? = null,
    val permissions: List<String> = emptyList(),
)
